src/nodedef.h

```diff
@@ -112,6 +112,28 @@ struct NodeBox
 struct MapNode;
 class NodeMetadata;
 
+enum LeavesStyle
+{
+	LEAVES_FANCY,
+	LEAVES_SIMPLE,
+	LEAVES_OPAQUE,
+};
+
+struct TextureSettings
```

Comment:
We use struct for plain-old-data types.  Objects with non-trivial constructors or any other methods qualify as a class, I believe.